File: Daily Challenge/C++/Merge Digit - Sort Integers 24.1.2026.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int N;
    cin >> N;

    vector<int> res;

    for (int i = 0; i < N; i += 2) {
        int a, b;
        cin >> a >> b;

        int ah = a / 100, at = (a / 10) % 10, ao = a % 10;
        int bh = b / 100, bt = (b / 10) % 10, bo = b % 10;

        int merged =
            max(ah, bh) * 100000 +
            min(ah, bh) * 10000 +
            max(at, bt) * 1000 +
            min(at, bt) * 100 +
            max(ao, bo) * 10 +
            min(ao, bo);

        res.push_back(merged);
    }

    sort(res.begin(), res.end());

    for (int x : res)
        cout << x << " ";
}

File: Daily Challenge/Python/Merge Digit - Sort Integers 24.1.2026.py
N = int(input())
nums = list(map(int, input().split()))

res = []

for i in range(0, N, 2):
    a, b = nums[i], nums[i+1]

    ah, at, ao = a // 100, (a // 10) % 10, a % 10
    bh, bt, bo = b // 100, (b // 10) % 10, b % 10

    merged = int(
        str(max(ah, bh)) + str(min(ah, bh)) +
        str(max(at, bt)) + str(min(at, bt)) +
        str(max(ao, bo)) + str(min(ao, bo))
    )

    res.append(merged)

res.sort()
print(*res)